Sources/SKSupport/Array+RemoveDuplicatesFromSorted.swift

```diff
@@ -0,0 +1,18 @@
+public extension Array where Element: Comparable {
+  /// Assuming that the array is sorted, remove all duplicate entries.
+  mutating func removeDuplicatesFromSorted() {
+    var lastElement: Element?
+    for (index, element) in zip(self.indices, self).reversed() {
+      if lastElement == element {
+        self.remove(at: index)
```

Comment:
This is quadratic since it needs to shift the trailing elements every time, and it requires an allocation since it captures self in the zip before modifying it.  You could use `removeAll(where:)` to do it in linear time, or else build a new array from the uniqued values instead of modifying the existing array in-place.

For the current use-case I expect this doesn't matter, since you're probably not removing many elements.  But I figured I would mention it since this is a generic method on Array.

Reply:
Very good catch. Another classical case of premature optimization I fell for. I thought this would be faster than `Set(ranges).sorted()` because we can harvest the property that the array is already sorted. But doing some measurements shows I’m totally wrong. `Set(ranges).sorted()` outperforms anything I tried and is way simpler.